app/src/utils/clients/ReportDbClient.ts

```diff
@@ -0,0 +1,187 @@
+import { IDBReport, IReportClient, IReport } from '@/models';
+import { AttributeValue, DynamoDB } from '@aws-sdk/client-dynamodb';
+import { getDynamoDbClient } from '../api';
+import { getUuid, isUndefined } from '../common';
+import { getStage, getTenant } from '../environment';
+
+const BASE_TABLE_NAME = 'ReportsTable';
+const CATEGORY_INDEX_NAME = 'CategoryIndex';
+
+/** Client to interact with report DynamoDB */
+export class ReportDbClient implements IReportClient {
+    ddbClient: DynamoDB;
+
+    constructor() {
+        this.ddbClient = getDynamoDbClient();
+    }
+
+    /** Create report */
+    async createReport(report: Omit<IReport, 'reportId'>) {
+        const reportId = getUuid();
+
+        await this.ddbClient.putItem({
+            TableName: getTableName(),
+            Item: marshalReport({
+                ...report,
+                reportId,
+            }),
+        });
+
+        return this.getReport(reportId);
+    }
+
+    /** Get report */
+    async getReport(reportId: string) {
+        const key: Omit<
```

Comment:
Should this use `Pick` instead of `Omit` so you don't need to itemize all the keys?